### PR TITLE
refactor: fix analyzer warnings

### PR DESCRIPTION
**Pull request by RushabhBhansali**
- IDE0120
- IDE0037
- IDE0071
- IDE0052
- IDE0032
- IDE0054
- SA1203
- IDE0037
- SA1316
- SA1216
- SA1401

### CODE REVIEW

File: src/Microsoft.ComponentDetection.Detectors/npm/NpmComponentUtilities.cs [hunk 1]
```diff
@@ -50,7 +50,7 @@ public static TypedComponent GetTypedComponent(JProperty currentDependency, stri
 
             if (!SemanticVersion.TryParse(version, out var result) && !TryParseNpmVersion(npmRegistryHost, name, version, out result))
             {
-                logger.LogInfo($"Version string {version} for component {name} is invalid or unsupported and a component will not be recorded.");
+                logger.LogInfo($"version string {version} for component {name} is invalid or unsupported and a component will not be recorded.");
```
Comment (JamieMagee) [inline src/Microsoft.ComponentDetection.Detectors/npm/NpmComponentUtilities.cs:53]: This looks like an unrelated change

File: src/Microsoft.ComponentDetection.Detectors/nuget/NuGetProjectModelProjectCentricComponentDetector.cs [hunk 2]
```diff
@@ -362,7 +362,7 @@ private LockFileLibrary GetLibraryComponentWithDependencyLookup(IList<LockFileLi
             LockFileLibrary matchingLibrary;
             if (version != null)
             {
-                // .Version.Version ensures we get to a nuget normalized 4 part version
+                // .version.version ensures we get to a nuget normalized 4 part version
```
Comment (JamieMagee) [inline src/Microsoft.ComponentDetection.Detectors/nuget/NuGetProjectModelProjectCentricComponentDetector.cs:365]: This looks like an unrelated change

File: src/Microsoft.ComponentDetection.Detectors/pip/PythonResolver.cs [hunk 3]
```diff
@@ -87,7 +87,7 @@ private async Task<IList<PipGraphNode>> ProcessQueue(PythonResolverState state)
                         // The currently selected version is invalid, try to see if there is another valid version available
                         if (!await this.InvalidateAndReprocessAsync(state, node, dependencyNode))
                         {
-                            this.Logger.LogWarning($"Version Resolution for {dependencyNode.Name} failed, assuming last valid version is used.");
+                            this.Logger.LogWarning($"version Resolution for {dependencyNode.Name} failed, assuming last valid version is used.");
```
Comment (JamieMagee) [inline src/Microsoft.ComponentDetection.Detectors/pip/PythonResolver.cs:90]: This looks like an unrelated change

File: src/Microsoft.ComponentDetection.Detectors/pip/PythonResolver.cs [hunk 4]
```diff
@@ -208,12 +208,12 @@ private void AddGraphNode(PythonResolverState state, PipGraphNode parent, string
...
         {
-            public readonly IDictionary<string, SortedDictionary<string, IList<PythonProjectRelease>>> ValidVersionMap
+            public IDictionary<string, SortedDictionary<string, IList<PythonProjectRelease>>> ValidVersionMap { get; }
                 = new Dictionary<string, SortedDictionary<string, IList<PythonProjectRelease>>>(StringComparer.OrdinalIgnoreCase);
 
-            public readonly Queue<(string, PipDependencySpecification)> ProcessingQueue = new Queue<(string, PipDependencySpecification)>();
+            public Queue<(string, PipDependencySpecification)> ProcessingQueue { get; } = new Queue<(string, PipDependencySpecification)>();
 
-            public readonly IDictionary<string, PipGraphNode> NodeReferences = new Dictionary<string, PipGraphNode>(StringComparer.OrdinalIgnoreCase);
+            public IDictionary<string, PipGraphNode> NodeReferences { get; } = new Dictionary<string, PipGraphNode>(StringComparer.OrdinalIgnoreCase);
```
Comment (JamieMagee) [inline src/Microsoft.ComponentDetection.Detectors/pip/PythonResolver.cs:216]: Why remove `readonly`?

File: src/Microsoft.ComponentDetection.Detectors/pip/PythonVersionUtilities.cs [hunk 5]
```diff
@@ -9,8 +9,8 @@ public static class PythonVersionUtilities
         /// <summary>
         /// Determine if the version is valid for all specs.
         /// </summary>
-        /// <param name="version">Version.</param>
-        /// <param name="specs">Version specifications.</param>
+        /// <param name="version">version.</param>
+        /// <param name="specs">version specifications.</param>
```
Comment (JamieMagee) [inline src/Microsoft.ComponentDetection.Detectors/pip/PythonVersionUtilities.cs:13]: This looks like an unrelated change

File: src/Microsoft.ComponentDetection.Detectors/rust/RustCrateDetector.cs [hunk 6]
```diff
@@ -19,7 +19,7 @@ public class RustCrateDetector : FileComponentDetector
     {
         private const string CargoLockSearchPattern = "Cargo.lock";
 
-        ////  PkgName[ Version][ (Source)]
+        ////  PkgName[ version][ (Source)]
```
Comment (JamieMagee) [inline src/Microsoft.ComponentDetection.Detectors/rust/RustCrateDetector.cs:22]: This looks like an unrelated change

File: src/Microsoft.ComponentDetection.Detectors/rust/SemVer/Comparator.cs [hunk 7]
```diff
@@ -120,6 +116,10 @@ public enum Operator
             GreaterThanOrEqual,
         }
 
+        public Operator ComparatorType { get; set; }
+
+        public SemVersion Version { get; set; }
+
```
Comment (JamieMagee) [inline src/Microsoft.ComponentDetection.Detectors/rust/SemVer/Comparator.cs:122]: `readonly`?

File: src/Microsoft.ComponentDetection.Detectors/yarn/YarnLockFile.cs [hunk 8]
```diff
@@ -5,7 +5,7 @@ namespace Microsoft.ComponentDetection.Detectors.Yarn
     public class YarnLockFile
     {
         /// <summary>
-        /// Gets or sets the declared Yarn Lock Version.
+        /// Gets or sets the declared Yarn Lock version.
```
Comment (JamieMagee) [inline src/Microsoft.ComponentDetection.Detectors/yarn/YarnLockFile.cs:8]: This looks like an unrelated change

File: src/Microsoft.ComponentDetection.Detectors/pip/PythonResolver.cs (hunk 4 above)
Comment (RushabhBhansali) [reply src/Microsoft.ComponentDetection.Detectors/pip/PythonResolver.cs:216]: it was necessary to convert field into a property to resolve [SA1401 : field should be private](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/documentation/SA1401.md). And read-only is not valid on a property.

File: src/Microsoft.ComponentDetection.Detectors/pip/PythonVersionUtilities.cs (hunk 5 above)
Comment (RushabhBhansali) [reply src/Microsoft.ComponentDetection.Detectors/pip/PythonVersionUtilities.cs:13]: since name of the parameter changed, we should also change the documentation ?

File: src/Microsoft.ComponentDetection.Detectors/rust/SemVer/Comparator.cs (hunk 7 above)
Comment (RushabhBhansali) [reply src/Microsoft.ComponentDetection.Detectors/rust/SemVer/Comparator.cs:122]: it was necessary to convert field into a property to resolve [SA1401 : field should be private](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/documentation/SA1401.md). And read-only is not valid on a property. 

However, we should not be providing setter for this property and I fixed that by removing the setter.
